calico_containers/pycalico/util.py

```diff
@@ -10,3 +26,43 @@ def generate_cali_interface_name(prefix, ep_id):
     if len(prefix) > 4:
         raise ValueError('Prefix must be 4 characters or less.')
     return prefix + ep_id[:11]
+
+
+def get_host_ips(version=4, interface=None, exclude=[]):
+    """
+    Gets all IP addresses assigned to this host.
+
+    This function is fail-safe and will return an empty array instead of
+    raising any exceptions.
+
+    :param version: Desired version of IP addresses. Can be 4 or 6. defaults to 4
+    :param interface: string - Desired interface to pull IP address from
+    :param exclude: list of interfaces (strings) to ignore (ex. ["lo","docker0"])
+    :return: List of string representations of IP Addresses.
+    """
+
+    ip_addrs = []
+
+    # Call `ip addr`
+    if interface is None:
+        ip_addr_output = check_output(["ip", "-%d" % (version), "addr"])    
+    else:    # Add argument to `ip addr` command if you are selecting an interface
+        ip_addr_output = check_output(["ip", "-%d" % (version), "addr", "show", "%s" % (interface)])
```

Comment:
What is the use case for providing the `interface=` argument?

Is there a reason we need this argument now?  It seems to me that it is never used.  If this is so, I think we should not include this argument.  It means less code to write, less code to maintain, less code to test.

If we ever need this functionality in the future, then we can add it at that point.